pkg/compilers/osv/osv_node.go

```diff
@@ -0,0 +1,143 @@
+package osv
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"github.com/emc-advanced-dev/pkg/errors"
+	"github.com/emc-advanced-dev/unik/pkg/types"
+	unikutil "github.com/emc-advanced-dev/unik/pkg/util"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// nodeProjectConfig defines available inputs
+type nodeProjectConfig struct {
+	dynamicProjectConfig
+	MainFile string `yaml:"main_file"`
+}
+
+type capstanMeta struct {
+	Name    string            `yaml:"name"`
+	Title   string            `yaml:"title"`
+	Author  string            `yaml:"author"`
+	Version string            `yaml:"version,omitempty"`
+	Require []string          `yaml:"require,omitempty"`
+	Binary  map[string]string `yaml:"binary,omitempty"`
+}
+
+type capstanRunNode struct {
+	Runtime string `yaml:"runtime"`
+	Main    string `yaml:"main"`
+}
+
+// CreateImageNode creates OSv image from project source directory and returns filepath of it.
+func CreateImageNode(params types.CompileImageParams, useEc2Bootstrap bool) (string, error) {
+	// Prepare meta files for capstan (based on manifest.yaml)
+	var config nodeProjectConfig
+	data, err := ioutil.ReadFile(filepath.Join(params.SourcesDir, "manifest.yaml"))
+	if err != nil {
+		return "", errors.New("failed to read manifest.yaml file", err)
+	}
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return "", errors.New("failed to parse yaml manifest.yaml file", err)
+	}
+	if err := generateCapstanMetaFilesNode(params.SourcesDir, &config); err != nil {
```

Comment:
since we're generating some files here that aren't necessary after this function returns, we should `defer` cleaning them up here (except in the case that CompileParams.NoCleanup == true)